core/corehttp/gateway_test.go

```diff
@@ -114,21 +130,27 @@ func TestGatewayGet(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	ns["/ipns/example.com"] = path.FromString("/ipfs/" + k)
+	kTTL := 10 * time.Minute
+	ns["/ipns/example.com"] = mockEntry{
+		path: path.FromString("/ipfs/" + k),
+		ttl:  kTTL,
+	}
 
 	t.Log(ts.URL)
 	for _, test := range []struct {
 		host   string
 		path   string
 		status int
+		ttl    []time.Duration // an approximation for a Maybe/Option sum type
```

Comment:
The entries in the table being generated here here need to include “no TTL”, `namesys.ImmutableTTL` and `kTTL` as values for this field. I could create a local copy of `namesys.ImmutableTTL` to be able to use a pointer here, but would that be any cleaner?